src/creature.cpp

```diff
@@ -846,13 +846,9 @@ BlockType_t Creature::blockHit(Creature* attacker, CombatType_t combatType, int3
 		if (checkArmor) {
 			int32_t armorValue = getArmor();
 			if (armorValue > 1) {
-				double armorFormula = armorValue * 0.475;
-				int32_t armorReduction = static_cast<int32_t>(std::ceil(armorFormula));
-				damage -= uniform_random(
-					armorReduction,
-					armorReduction + static_cast<int32_t>(std::floor(armorFormula))
-				);
-			} else if (armorValue == 1) {
+				int32_t maxArmorReduction = (armorValue % 2) == 0 ? armorValue - 1 : armorValue - 2;
+				damage -= uniform_random(static_cast<int32_t>(std::floor(armorValue / 2.)), maxArmorReduction);
```

Comment:
Instead of `static_cast<int32_t>(std::floor(armorValue / 2.)`, you can use `armorValue >> 1`, which will perform flooring too and is simpler.